vignettes/articles/managing-tokens-securely.Rmd

```diff
@@ -0,0 +1,198 @@
+---
+title: "Managing tokens securely"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Managing tokens securely}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+**DRAFT!**
+
+Testing presents special challenges for packages that wrap an API. Here we tackle one of those problems: how to deal with auth in a non-interactive setting on a remote machine. This affects gargle itself and will affect any client package that relies on gargle for auth.
+
+This article documents the token management approach taken in gargle. We wanted it to be relatively easy to have a secret, such as an auth token, that we can:
+
+  * Use locally
+  * Use with continuous integration (CI) services, such as Travis-CI
+  * Use with [R-hub](https://docs.r-hub.io)
+
+all while keeping the secret secure.
+
+The approach uses symmetric encryption, where the shared key is stored in an environment variable. Why? This works well with existing conventions for local R usage. Most CI services offer support for secure environment variables. And R-hub accepts environment variables via the `env_vars` argument of `rhub::check()`.
+
+This is based on an approach originally worked out in [bigrquery](https://bigrquery.r-dbi.org).
+
+## Accessing the `secret_*()` functions
+
+gargle's approach to managing test tokens is implemented through several functions that all start with the `secret_` prefix. These functions are not (currently?) exported. This may seem odd, since others might want to use these functions. But note they are only needed during setup or at test time. This sort of usage is compatible with others calling internal gargle functions and possibly inlining a version of one or two test helpers.
+
+One way to make the `secret_*()` functions available for local experimentation is to call `devtools::load_all()`, which exposes all internal objects in a package:
+
+```{r eval = FALSE}
+devtools::load_all("path/to/source/of/gargle/")
+```
+
+The approach I'll take in this article is to call these functions via `:::`.
+
+We go ahead and attach gargle in the normal way, which makes the exported functions readily available.
+
+```{r setup}
+library(gargle)
+```
+
+## Overview of the approach
+
+  * Generate a random PASSWORD and give it a self-documenting name, e.g. `GARGLE_PASSWORD`.
```

Comment:
Here you make it sound like a naming suggestion, but it looks like your functions require the format `<package>_PASSWORD`

Reply:
True. Reworded.